### PR TITLE
[SMALLFIX] further unifying master and worker

### PR DESCRIPTION
**Pull request by jsimsa**

### CODE REVIEW

File: core/server/src/main/java/alluxio/worker/AlluxioWorkerService.java [hunk 1]
```diff
@@ -12,17 +12,53 @@
...
  * A worker in the Alluxio system.
  */
 public interface AlluxioWorkerService extends Server {
   /**
+   * Factory for creating {@link AlluxioMasterService}.
+   */
+  @ThreadSafe
+  final class Factory {
+    /** The worker service loaders. */
+    private static ServiceLoader<WorkerFactory> sServiceLoader;
```
Comment (aaudiber) [inline core/server/src/main/java/alluxio/worker/AlluxioWorkerService.java]: Does caching this give a noticeable speedup to test times?

File: core/server/src/main/java/alluxio/worker/AlluxioWorkerService.java [hunk 2]
```diff
@@ -12,17 +12,53 @@
...
+     * @return a new instance of {@link AlluxioWorkerService}
+     */
+    public static AlluxioWorkerService create() {
+      return new DefaultAlluxioWorker();
+    }
+
+    /**
+     * @return the (cached) worker service loader
+     */
+    static synchronized ServiceLoader<WorkerFactory> getServiceLoader() {
```
Comment (aaudiber) [inline core/server/src/main/java/alluxio/worker/AlluxioWorkerService.java]: This method seems like an implementation detail, not something that a user of the class cares about. Would it make more sense to put it in the concrete class?

File: core/server/src/main/java/alluxio/worker/AlluxioWorkerService.java (hunk 1 above)
Comment (jsimsa) [inline core/server/src/main/java/alluxio/worker/AlluxioWorkerService.java]: not sure, but we do it for the master

File: core/server/src/main/java/alluxio/worker/AlluxioWorkerService.java (hunk 2 above)
Comment (jsimsa) [inline core/server/src/main/java/alluxio/worker/AlluxioWorkerService.java]: I moved the logic to ServerUtils

File: core/server/src/main/java/alluxio/worker/AlluxioWorkerService.java (hunk 1 above)
Comment (aaudiber) [inline core/server/src/main/java/alluxio/worker/AlluxioWorkerService.java]: I'd prefer to avoid static state if we can help it, up to you

**Comment by apc999** [comment]
LGTM

**Pull request merged**